library/src/main/java/com/android/volley/RequestQueue.java

```diff
@@ -343,6 +356,8 @@ void finish(Request<?> request) {
                 }
             }
         }
+
+        mRequestLogger.log(request);
```

Comment:
does it not make sense to have the logic located where the request is added to the queue. Not when a queue is given to the request. I know the request has some of its own logic around its state and has a `finish` method, but I think we want to move to a model where a request is just a dumb object.